### DISCUSSION TITLE
Unable To Use [pre-entrypoint,entrypoint,post-entrypoint] in action.yml

### DISCUSSION THREAD

**Discussion by varunsridharan** [community/community#26429] [Actions]
<p>After reading the docs @ <a href="https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/metadata-syntax-for-github-actions#pre-entrypoint" rel="noopener nofollow ugc">https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/metadata-syntax-for-github-actions#pre-entrypoint</a></p>
<p>I tried to use <code>pre-entrypoint</code> &amp; <code>entrypoint</code> in my github action but i keep getting error like below</p>
<pre><code class="lang-auto">docker: Error response from daemon: OCI runtime create failed: container_linux.go:370: starting container process caused: exec: "entrypoint2.sh": executable file not found in $PATH: unknown.
</code></pre>
<h3>My current action.yml looks like below</h3>
<pre><code class="lang-auto">name: 'Action Test'
description: ""
author: "varunsridharan"
runs:
  using: 'docker'
  image: 'docker://ghcr.io/varunsridharan/actions-alpine-php:latest'
  entrypoint: 'entrypoint2.sh'
</code></pre>

**Comment by airtower-luna** [comment, marked as answer]
<p>There are a number of things that could be causing this:</p>
<ul>
<li>The <code>entrypoint2.sh</code> script is missing from the Docker image</li>
<li>The <code>entrypoint2.sh</code> script is not marked as executable (use <code>chmod a+x</code> to change that)</li>
<li>The path to <code>entrypoint2.sh</code> is missing in the <code>entrypoint:</code> parameter. Without specifying a full path the script must be in one of the directories listed in <code>$PATH</code>. For example, if the script is in the root of the container image, you’d need to use:</li>
</ul>
<pre><code class="lang-yaml">  entrypoint: '/entrypoint2.sh'
</code></pre>

**Reply by matifali** [reply]
Should I make `entrypoint.sh` executable inside my docker image even if I pass it as metadata in `action.yaml`?

**Reply by airtower-luna** [reply]
Yes, otherwise it can't be executed.

**Reply by matifali** [reply]
My question is as I am not copying `entrypoint.sh` in my Dockerfile, (only passing it as metadata)
So how and where should I set it as executable?

**Reply by airtower-luna** [reply]
"Only as metadata" won't work at all, unless you mean selecting something already present in the base image. If you mean it's in a volume mount (e.g. the workspace) it needs to be set as executable there.

**Reply by matifali** [reply]
Thanks.
I get it.
